### PR TITLE
NurseConnect Move ID data collection 

### PR DESCRIPTION
**Pull request by AyabongaQwabi**
Collection of passport/id information should not be during registration process but in change app

### CODE REVIEW

File: go-app-nurse_ussd.js [hunk 1]
```diff
@@ -1538,12 +1538,14 @@ go.app = function() {
     // INITIAL STATES
 
         self.add('st_subscribed', function(name) {
+
```
Comment (gsvr) [inline go-app-nurse_ussd.js]: remove unnecessary whitespace

File: go-app-nurse_ussd.js [hunk 2]
```diff
@@ -1553,7 +1555,7 @@ go.app = function() {
                 }
             });
         });
-
+
```
Comment (gsvr) [inline go-app-nurse_ussd.js]: set delete blank whitespace option in sublime

File: test/nurse_ussd.test.js [hunk 3]
```diff
@@ -243,7 +243,7 @@ describe("app", function() {
                 });
             });
             describe("registered user", function() {
-                it("should give 6 options", function() {
+                it.skip("should give 6 options", function() {
```
Comment (gsvr) [inline test/nurse_ussd.test.js]: Update this to show 7 options.  Use a PaginatedChoiceState for the state to prevent char overflow.

File: src/nurse_ussd.js [hunk 4]
```diff
@@ -175,16 +176,21 @@ go.app = function() {
...
                     new Choice('st_subscribe_other', $('Subscribe a friend')),
                     new Choice('st_change_num', $('Change your no.')),
                     new Choice('st_change_faccode', $('Change facility code')),
+                    new Choice('st_change_id_no', $('Change ID no.')),
                     new Choice('st_change_sanc', $('Change SANC no.')),
                     new Choice('st_change_persal', $('Change Persal no.')),
                     new Choice('isl_check_optout_optout', $('Stop SMS')),
                 ],
+                characters_per_page: 140,
+                options_per_page: 4,
```
Comment (gsvr) [inline src/nurse_ussd.js]: Set `options_per_page` to `null` and just comment it out since it defaults to `null`.  I like to leave it in the code as a comment so it's easy to change later.  When it's `null` the State calculates how many options to display, which is what we want.

File: src/nurse_ussd.js [hunk 5]
```diff
@@ -175,16 +176,21 @@ go.app = function() {
...
+                more:$('more'),
+                back:$('back'),
```
Comment (gsvr) [inline src/nurse_ussd.js]: Please capitalize the quoted strings to be `More` and `Back`.
Also, add a space after each `:`

File: src/nurse_ussd.js [hunk 6]
```diff
@@ -258,6 +264,93 @@ go.app = function() {
...
+            return new ChoiceState(name, {
+                question: $("Please select {{owner}} type of identification:")
+                    .context({owner: owner}),
+                choices: [
+                    new Choice('st_id_no', $('RSA ID')),
+                    new Choice('st_passport', $('Passport'))
+                ],
+                next: function(choice) {
+                    return choice.value;
+
```
Comment (gsvr) [inline src/nurse_ussd.js]: Remove this blank line please.

File: src/nurse_ussd.js [hunk 7]
```diff
@@ -258,6 +264,90 @@ go.app = function() {
...
+                    new Choice('ls', $('Lesotho')),
+                    new Choice('cu', $('Cuba')),
+                    new Choice('other', $('Other')),
+                ],
+                next: 'st_passport_no'
+            });
+        });
+
+        self.add('st_passport_no', function(name) {
+            return new FreeText(name, {
```
Comment (gsvr) [inline src/nurse_ussd.js:328]: What happened to the `var error` and `var question` parts?  I suggest copying the whole `self.add` code block again from the original and just updating the state names.

File: src/nurse_ussd.js [hunk 8]
```diff
@@ -258,6 +264,90 @@ go.app = function() {
...
+                        return error;
+                    }
+                },
+                next: 'st_passport_dob'
+            });
+        });
+
+        self.add('st_passport_dob', function(name) {
+            return new FreeText(name, {
+                question: $('please enter the pthe date of birth e.g. 27 May 1975 as 27051975:'),
```
Comment (gsvr) [inline src/nurse_ussd.js]: This text is not the same as the original text.  It also needs the `var question` and `var error` bits.  The `if` is also indented strangely.  Please copy the whole `self.add` block again and change the state names.

File: test/nurse_ussd.test.js [hunk 9]
```diff
@@ -257,14 +257,56 @@ describe("app", function() {
...
-                                '6. Stop SMS'
+                                '4. Change ID no.',
+                                '5. Change SANC no.',
+                                '6. More'
                             ].join('\n')
                         })
                         .run();
                 });
-            });
+                it("should give 4 options when user selects more", function() {
```
Comment (gsvr) [inline test/nurse_ussd.test.js]: `2 options`

File: test/nurse_ussd.test.js [hunk 10]
```diff
@@ -257,14 +257,56 @@ describe("app", function() {
...
+                            reply: [
+                                "Welcome to NurseConnect",
+                                "1. Change Persal no.",
+                                "2. Stop SMS",
+                                "3. Back"
+                            ].join('\n')
+                        })
+                        .run();
+                });
+                it("should give the first 6 options when user selects back", function() {
```
Comment (gsvr) [inline test/nurse_ussd.test.js]: `5 options`

File: test/nurse_ussd.test.js [hunk 11]
```diff
@@ -257,14 +257,56 @@ describe("app", function() {
...
                         })
                         .run();
                 });
-            });
+                it("should give 4 options when user selects more", function() {
+                    return tester
+                        .setup.user.addr('27821237777')
+                        .setup.char_limit(140)  // limit first state chars
+                        .inputs(
+                            {session_event: 'new'},  // dial in
```
Comment (gsvr) [inline test/nurse_ussd.test.js]: Put the comma at the end of this line at the beginnning of the next line. This helps a lot with multiple cursor usage.  (Apply this everywhere in the `inputs`)

File: test/nurse_ussd.test.js [hunk 12]
```diff
@@ -429,6 +471,83 @@ describe("app", function() {
...
+            it("Should ask for their ID no",function(){
+                return tester
+                    .setup.user.addr('27821237777')
+                    .inputs(
+                            {session_event:'new'}  // dial in
+                            , '4'  // st_subscribed - change id
+                            , '1'  // st_change_id_no - RSA ID
+                    )
+                    .check.interaction({
+                        state:'st_id_no',
```
Comment (gsvr) [inline test/nurse_ussd.test.js]: Put spaces after `:`

File: test/nurse_ussd.test.js [hunk 13]
```diff
@@ -429,6 +471,83 @@ describe("app", function() {
...
+                    .setup.user.addr('27821237777')
+                    .inputs(
+                            {session_event:'new'}  // dial in
+                            , '4'  // st_subscribed - change id
+                            , '2'  // st_change_id_no - Passport
+                    )
+                    .check.interaction({
+                        state:'st_passport',
+                        reply:[
+                            'What is the country of origin of the passport',
```
Comment (gsvr) [inline test/nurse_ussd.test.js]: Question mark missing at the end of the string.

**Comment by gsvr** [comment]
@imsickofmaps Please review.

**Comment by imsickofmaps** [comment]
👍  🎉
